pkg/patterns/declarative/reconciler.go

```diff
@@ -74,6 +74,14 @@ type DeclarativeObject interface {
 	metav1.Object
 }
 
+type Pruner interface {
```

Comment:
Could you add some explaination for this interface?

Reply:
An explanation of the interface has been appended.